app/controllers/sites_controller.rb

```diff
@@ -1,8 +1,62 @@
 class SitesController < ApplicationController
-  before_action :authenticate
+  before_action :authenticate, :build_provider, :initialise_errors
 
   def index
-    @provider = Provider.includes(:sites).find(params[:provider_code]).first
     @sites = @provider.sites.sort_by(&:location_name)
   end
+
+  def edit
+    @site = @provider.sites.find { |site| site.id == params[:id] }
```

Comment:
MINOR: Feels like it'd be worth putting this into a builder hook for `edit` and `update`.

Reply:
Done